src/java.base/share/classes/java/lang/reflect/AccessFlag.java

```diff
@@ -385,6 +384,30 @@ public static Set<AccessFlag> maskToAccessFlags(int mask, Location location) {
         return new AccessFlagSet(definition, mask);
     }
 
+    /**
+     * {@return an unmodifiable set of access flags for the given mask value
+     * appropriate for the location in the given class file format version}
+     *
+     * @param mask bit mask of access flags
+     * @param location context to interpret mask value
+     * @param cffv the class file format to interpret mask value
+     * @throws IllegalArgumentException if the mask contains bit
+     * positions not defined for the location in the given class file format
+     * @throws NullPointerException if {@code location} or {@code cffv} is {@code null}
+     * @since 25
+     */
+    public static Set<AccessFlag> maskToAccessFlags(int mask, Location location, ClassFileFormatVersion cffv) {
+        var definition = findDefinition(location);
+        int unmatchedMask = mask & (~location.flagsMask(cffv)); // implicit null check
```

Comment:
Implicit null check is actually in `findDefinition`.
There's little harm is looking up the definition before checking for unmatched mask bits but it could be delayed and inlined as the argument to `new AccessFlagSet`.

Reply:
Oops, this was the implicit null check for cffv. Should I make all of the checks explicit?

Also, I split out `findDefinition` as in value objects, it will become `findDefinition(location, cffv)`, when the valhalla cffv is passed, findDefinition can return differently for Location.CLASS.

Reply:
Explicit null checks are better if they provide if they name the caller supplied argument that was null.
Checking for caller errors usually precedes the methods semantics.